tagging/utils.py

```diff
@@ -3,12 +3,25 @@
 calculation.
 """
 import math
-import types
 
 from django.db.models.query import QuerySet
-from django.utils.encoding import force_unicode
+try:
+    from django.utils.encoding import force_unicode
+except ImportError:
+    def force_unicode(s, *args, **kwargs):
+        return s
 from django.utils.translation import ugettext as _
 
+try:
+    basestring
+except NameError:  # Python 3 doesn't have basestring
+    basestring = str
+
+try:
+    long
+except NameError:  # Python 3 doesn't have long
+    long = int
```

Comment:
oops.  I had a mixup when trying to parse what was actually in 3.x and what was in 2.x

For this, since your tests pass already, I think you can leave it as is.  If there were more instances of things like this (for some other app) I'd be more for changing it